bridges/otellogr/logsink.go

```diff
@@ -0,0 +1,403 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otellogr provides [LogSink], an [logr.LogSink] implementation, that
+// can be used to bridge between the [github.com/go-logr/logr] API and
+// [OpenTelemetry].
+//
+// # Record Conversion
+//
+// The logr records are converted to OpenTelemetry [log.Record] in the following
+// way:
+//
+//   - Time is set as the current time of conversion.
+//   - Message is set as the Body using a [log.StringValue].
+//   - Level is transformed and set as the Severity. The SeverityText is not
+//     set.
+//   - PC is dropped.
+//   - KeyAndValues are transformed and set as Attributes.
+//   - Error is always logged as an additional attribute with the key "err" and
+//     with the severity [log.SeverityError].
+//   - Name is logged as an additional attribute with the key "logger".
+//   - Context is not propagated to the OpenTelemetry log record, as logr does
+//     not provide a way to access it.
```

Comment:
Can we handle it if someone passes context as an attributes? 

This is what I have done in https://github.com/pellared/opentelemetry-go/blob/147762ffb5c4394417b3f973274cc570723db79b/log/internal/logr.go.